internal/shortener/service.go

```diff
@@ -0,0 +1,101 @@
+package shortener
+
+import (
+	"fmt"
+	"github.com/Azzonya/go-shortener/internal/entities"
+	"github.com/Azzonya/go-shortener/internal/repo"
+	"github.com/Azzonya/go-shortener/internal/storage"
+	"github.com/google/uuid"
+	"net/url"
+)
+
+type Shortener struct {
+	repo    repo.Repo
+	storage *storage.Storage
+	baseURL string
+	UseDB   bool
+}
+
+func New(baseURL string, storage *storage.Storage, repo repo.Repo, UseDB bool) *Shortener {
+	return &Shortener{
+		baseURL: baseURL,
+		storage: storage,
+		repo:    repo,
+		UseDB:   UseDB,
+	}
+}
+
+func (s *Shortener) GetOneByShortURL(key string) (string, bool) {
+	var URL string
+	var exist bool
+
+	if !s.UseDB {
```

Comment:
```suggestion
	if s.UseDB {
```
понятней будет через утверждение